python_modules/dagster/dagster/_core/blueprints/shell_command_asset_blueprint.py

```diff
@@ -0,0 +1,61 @@
+from typing import AbstractSet, Literal, Mapping, Optional, Sequence, Union
+
+from dagster import AssetExecutionContext, MaterializeResult
+from dagster._core.blueprints.base_asset_blueprint import (
+    BaseAssetBlueprint,
+    BaseMultiAssetBlueprint,
+)
+
+
+class ShellCommandAssetBlueprint(BaseAssetBlueprint):
```

Comment:
Nice - out of curiosity, does this work with plan-ol-shell-scripts which don't use pipes?

Trying to think of a way in which we could make the client optional / conditionally create one for the user but I don't think there's a nice way to do so.

Reply:
My goal was to make it to.

This is what the interior of `PipesSubprocessClient.run` looks like:

```python
            process = Popen(
                command,
                cwd=cwd or self.cwd,
                env={
                    **os.environ,
                    **self.env,
                    **(env or {}),
                    **pipes_session.get_bootstrap_env_vars(),
                },
            )
```

So the only difference from launching a plain old shell script is that we add some env vars.

I'm realizing now that `get_materialize_result` might fail if the script doesn't send back a `MaterializeResult`. Will update the PR to not fail in that case.

Reply:
oh nvm the default `implicit_materialization=True` should solve this, will verify this with a test before taking this out of RFC